lisp/ess-custom.el

```diff
@@ -646,211 +646,345 @@ regardless of where in the line point is when the TAB command is used."
+  object <- some_function(other_function(
+                              arg1,
+                              arg2)
 
-Set `ess-arg-function-offset' to nil if you want:
 
-  some.function(arg1,
-                arg2=other.function(a,
-                                    b,
+If a list of the form '(N) where N is a number, the statements
+are indented at the previous line indentation + N characters:
+
+  object <- some_function(other_function(
+      arg1,
+      arg2)
 
-and
 
-some.function(arg1,
-              arg2=other.function(
-                     a,
-                     b,
+If nil, the statements are indented at the opening delimiter
+following foo:
 
+  object <- some_function(other_function(
+                                         arg1,
+                                         arg2)
+
+
+You can refer to ess-indent-level by setting this parameter to t
+or '(t) instead of N or '(N).
 ")
 
-;;added rmh 2Nov97 at request of Terry Therneau
-(defvar ess-close-brace-offset 0
-  "Extra indentation for closing braces.")
-
-(defvar ess-close-paren-offset 0
-  "Extra indentation for closing parenthesis.
-When a number, adjustment is made with respect to the opening
-parenthesis taking into account the value of
-`ess-arg-function-offset-new-line'. When set to 0 the closing
-parenthesis is indented as follows:
-
-some.function(arg1,
-              arg2
-              )
-
-When this variable is a list of the form '(N), N being a number,
-adjust with respect to the indentation of the line containing an
-opening parenthesis. For example, when set to '(0), the closing
-parenthesis is indented as follows:
-
-{
-    some.function(arg1,
-                  arg2 = X
-    )
-}
+(defvar ess-offset-arguments-closing 0
+  "Extra indentation for delimiters closing an expression.
+You can refer to ess-indent-level by setting this parameter to t.")
+
+(defvar ess-align-function-declaration t
+  "When this setting is switched on, function declarations ignore
+the offsets for function arguments and align all arguments from
+the opening parenthesis.
+
+`ess-offset-arguments' is set to 4 here:
+
+  some_function(argument1,
+      argument2, argument3
+  )
+
+But this offset is ignored for function declarations;
+
+  fun <- function(argument1,
+                  argument2
+                  argument3) {
+      body
+  }
+")
+
+(defvar ess-align-from-outer-parameter nil
+  "When a function is called as argument in another function
+call, should indentation start from the argument name to which
+the inner call is bound, or from the name of the inner function
+call itself?
+
+Only has an effect when the ess-offset-arguments is set to a
+number N (as opposed to nil or '(N))
+
+If nil:
+
+  some_function(parameter = other_function(
+                                argument
+                            ))
+
+If t:
+  some_function(parameter = other_function(
+                    argument
+                ))
 ")
 
+(defvar ess-offset-statement-continued-opening 0
```

Comment:
Why do we need this? I was hoping to simplify stuff as much as we can without losing flexibility. But it seems that in the new scheme there are even more configuration variables than before.